src/views/AuthView.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useUserContext } from '../context/UserContext';
+import { signInUser, signUpUser } from '../services/user';
+
+export default function AuthView() {
+  const { setCurrentUser, userId, setUserId } = useUserContext();
+  const [password, setPassword] = useState('');
+  const [email, setEmail] = useState('');
+  const [authType, setAuthType] = useState('login');
+  const [error, setError] = useState('');
+  const history = useHistory();
+
+  const handleSubmit = async () => {
+    try {
+      if (authType === 'login') {
+        const data = await signInUser({ email, password });
+        setCurrentUser(data);
+        setUserId(data.id);
+        history.push('/');
+
+      } else {
+        const data = await signUpUser({ email, password });
+        setCurrentUser(data.email);
```

Comment:
It appears that our signup route does not set the current user in state. Consider removing the .email here :)